### ISSUE TITLE
Enable SSI by default?

### ISSUE THREAD

**Issue by khalwat** [laravel/valet#513]
I have a number of websites that use [Nginx SSI](http://nginx.org/en/docs/http/ngx_http_ssi_module.html) (server side includes). Is there a chance that the default Valet Nginx config could have:

```
ssi on;
```

In the directives? Seems like it'd be a nice thing to have enabled by default in `local` dev

**Comment by drbyte** [comment]
In case this suggestion is not deemed something that gets merged, one can easily set this up for themselves:

Simply manually edit either or both of these files to add `ssi on;` in the position you wish (whether that's for the `server` or `location` blocks:
`~/.composer/vendor/laravel/valet/cli/stubs/secure.valet.conf`
`~/.composer/vendor/laravel/valet/cli/stubs/valet.conf`
This will take effect in all your configs if you run `valet domain test` (replace `test` here with whatever TLD you're using with valet), as the `valet domain` command will rebuild all your configs using those stubs.

Alternatively you could do it in `~/.composer/vendor/laravel/valet/cli/stubs/nginx.conf` 
and run `valet stop && valet remove && valet install`
... or manually edit `/usr/local/etc/nginx/valet/valet.conf` and restart nginx.

Whichever of these approaches you take, you'll need to re-edit the stubs manually anytime you upgrade valet to a newer version.

**Comment by khalwat** [comment]
Thanks @drbyte -- we do have this up and running locally, just hoping to get it merged into the defaults setup so that we don't have to repeat the setup for each developer/environment.

**Comment by khalwat** [comment]
So a quick explanation of [SSI](http://nginx.org/en/docs/http/ngx_http_ssi_module.html). SSI is Server Side Includes, which allows the server to do some (albeit limited) processing on the files that it serves.

Apache has an SSI module, and Nginx [has SSI baked in](http://nginx.org/en/docs/http/ngx_http_ssi_module.html).

To enable it in Nginx, the following directive is needed:

```
ssi on;
```

...in any of the following blocks: `http, server, location`

The default is `ssi off;` because if you're not using it, there's no reason to have it enabled and the (nominal) overhead that comes with it to parse for SSI statements.

SSI directives in Nginx are just specially formatted HTML comments, so they don't affect any browser rendering if somehow they are served up unprocessed.

So we can do things like change what is returned in the response based on the setting of a cookie:

```
<!--# if expr="$HTTP_COOKIE=/fonts\-loaded\=1/" -->
<html class="fonts-loaded" lang="{{ craft.app.language |slice(0,2) }}" prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb#">
<!--# else -->
<html lang="{{ craft.app.language |slice(0,2) }}" prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb#">
<!--# endif -->
```

It allows you to change the response based on certain conditions like the setting of a cookie, and more. This is extremely useful when you're using a static cache like FastCGI Cache where a backend templating language like Twig or Blade is never run.

Having this enabled by default in `local` development environments like Valet has no downside, because if you're not using SSI, nothing changes. But it will allow those of use who are doing SSI dev work to use Valet out of the box.

Which would be nice.

**Comment by khalwat** [comment]
https://github.com/laravel/valet/pull/517

**Comment by drbyte** [comment]
Can be closed now that #517 has been merged.
